tests/e2e/mapper/mapper_suite_test.go

```diff
@@ -0,0 +1,246 @@
+package mapper
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/api/core/v1"
+
+	"github.com/kubeedge/kubeedge/tests/e2e/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//context to load config and access across the package
+var (
+	ctx            *utils.TestContext
+	cfg            utils.Config
+	nodeSelector   string
+	nodeName       string
+	chconfigmapRet = make(chan error)
+)
+
+var (
+	runController       = "cd ${GOPATH}/src/github.com/kubeedge/kubeedge/cloud; sudo nohup ./edgecontroller > edgecontroller.log 2>&1 &"
+	runEdgecore         = "cd ${GOPATH}/src/github.com/kubeedge/kubeedge/edge/; sudo nohup ./edge_core > edge_core.log 2>&1 &"
+	ConfigmapHandler    = "/api/v1/namespaces/default/configmaps"
+	MockHandler         = "/apis/devices.kubeedge.io/v1alpha1/namespaces/default/devicemodels"
+	MockInstanceHandler = "/apis/devices.kubeedge.io/v1alpha1/namespaces/default/devices"
+	CrdHandler          = "/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions"
+	AppHandler          = "/api/v1/namespaces/default/pods"
+	NodeHandler         = "/api/v1/nodes"
+	DeploymentHandler   = "/apis/apps/v1/namespaces/default/deployments"
+	deviceCRD           = "devices.devices.kubeedge.io"
+	deviceModelCRD      = "devicemodels.devices.kubeedge.io"
+)
+
+type Token interface {
+	Wait() bool
+	WaitTimeout(time.Duration) bool
+	Error() error
+}
+
+func TestMapperCharacteristics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	var _ = BeforeSuite(func() {
+		utils.InfoV6("Before Suite Execution")
+		cfg = utils.LoadConfig()
+		ctx = utils.NewTestContext(cfg)
+		t := time.Now()
+		//nodeName = "integration-node-" + utils.GetRandomString(10)
+		nodeName = "edge-node1"
```

Comment:
Currently we wont be running these tests in CI, but later we may need to randomly create the edge-node name. Please add a TODO comment as well as create an issue.